File: JobTrigger-Backend/models/AppInfo.js
const mongoose = require('mongoose');

const appInfoSchema = new mongoose.Schema({
  appVersion: {
    type: String,
    required: true,
  },
  buildNumber: {
    type: String,
    required: true,
  },
  privacyPolicyUrl: {
    type: String,
    required: true,
  },
  termsOfServiceUrl: {
    type: String,
    required: true,
  },
  supportEmail: {
    type: String,
    required: true,
  },
  openSourceLicensesUrl: {
    type: String,
    required: true,
  }
}, {
  timestamps: true
});

module.exports = mongoose.model('AppInfo', appInfoSchema);

File: JobTrigger-Frontend/lib/domain/app_info/app_info.dart
/// Pure domain entity — mirrors the old Swift app's `AppInfo` model. `id`,
/// `createdAt`/`updatedAt` are storage bookkeeping only and deliberately not
/// carried into the domain layer (nothing in the UI needs them).
class AppInfo {
  const AppInfo({
    required this.appVersion,
    required this.buildNumber,
    this.privacyPolicyUrl,
    this.termsOfServiceUrl,
    this.supportEmail,
    this.openSourceLicensesUrl,
  });

  final String appVersion;
  final String buildNumber;
  final String? privacyPolicyUrl;
  final String? termsOfServiceUrl;
  final String? supportEmail;
  final String? openSourceLicensesUrl;
}
